aztec/src/main/kotlin/org/wordpress/aztec/source/InlineCssStyleFormatter.kt

```diff
@@ -36,21 +33,66 @@ class InlineCssStyleFormatter {
          * @param [start] The index where the [IAztecAttributedSpan] starts inside the [text].
          */
         fun applyInlineStyleAttributes(text: Editable, attributes: AztecAttributes, start: Int, end: Int) {
-            if (attributes.hasAttribute("style")) {
+            if (attributes.hasAttribute(STYLE_ATTRIBUTE)) {
 
                 if (start != end) {
-                    val style = attributes.getValue("", "style")
                     // Process the CSS 'color' property, remove any whitespace or newline characters
-                    val m = foregroundColorPattern.matcher(style.replace("\\s".toRegex(), ""))
-                    if (m.find()) {
-                        val colorString = m.group(1)
-                        val colorInt = ColorConverter.getColorInt(colorString)
-                        if (colorInt != ColorConverter.COLOR_NOT_FOUND) {
-                            text.setSpan(ForegroundColorSpan(colorInt), start, end, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
-                        }
-                    }
+                    val style = attributes.getValue("", STYLE_ATTRIBUTE).replace("\\s".toRegex(), "")
+
+                    processColor(style, text, start, end)
+                }
+            }
+        }
+
+        private fun getPattern(styleAttr: String): Pattern {
+            return Pattern.compile(
+                    "(?:;|\\A)$styleAttr:(.+?)(?:;|$)", Pattern.CASE_INSENSITIVE or Pattern.MULTILINE)
+        }
+
+        private fun processColor(style: String, text: Editable, start: Int, end: Int) {
+            val m = getPattern(CSS_COLOR_ATTRIBUTE).matcher(style)
```

Comment:
What do you thin if we move `getPattern(CSS_COLOR_ATTRIBUTE)`, and the other call `getPattern(CSS_TEXT_DECORATION_ATTRIBUTE)`, to a static variable in this class?
I see those methods called many times when I pass a text that does contain `style` attribute.

Reply:
What about making it more general - I could I add a method for retrieving style attributes `getStyleAttribute(attributeName)` so you don't have to deal with regex matching? I'd also change `containsUnderlineDecorationStyle()` to a more general public `containsStyle(styleAttr): Boolean`.

Reply:
That seems a nice idea!